### ISSUE TITLE
Left-brackets '[' being stripped from NXOS output in Netmiko '2.3.0'

### ISSUE THREAD

**Issue by jesse-mcgraw** [ktbyers/netmiko#1100]
[ characters are being removed from NXOS output and replaced with CR/LF

For example, expected output:

```
IP Route Table for VRF "test"
'*' denotes best ucast next-hop
'**' denotes best mcast next-hop
'[x/y]' denotes [preference/metric]
'%<string>' in via output denotes VRF <string>

10.0.0.1/32, ubest/mbest: 1/0
    *via 10.0.0.101, Po0.1, [115/6], 3d08h, isis-65535, L2

```

becomes

```
IP Route Table for VRF "test"
'*' denotes best ucast next-hop
'**' denotes best mcast next-hop
'
x/y]' denotes 
preference/metric]
'%<string>' in via output denotes VRF <string>

10.0.0.1/32, ubest/mbest: 1/0
    *via 10.0.0.101, Po0.1, 
115/6], 3d08h, isis-65535, L2
```

**Comment by ktbyers** [comment]
Can you show your connection arguments that you are using for the Netmiko connection?

**Comment by jesse-mcgraw** [comment]
Nothing too special:

`with ConnectHandler(fast_cli=False, **netmiko_parameters) as conn:`

and

```
- netmiko_parameters:
    host: {{ ROUTER.NAME }}
    device_type: cisco_nxos
    username: {{ user }}
    password: {{ password }}
    global_delay_factor: 1
```

**Comment by ktbyers** [comment]
Does the problem go away if you increase the global_delay_factor to 2 or 4.

Also would be interesting to see a print out of the output from Netmiko using a repr i.e.

```
print(repr(output))
```

To see if there are any ANSI escape codes in the output.

**Comment by jesse-mcgraw** [comment]
Changing global_delay_factor to 4 had no effect

Here is the print(repr(outout)) when switching the driver between cisco_ios and cisco_nxos and changing nothing else:


cisco_ios:
```
'IP Route Table for VRF "test"\n\'*\' denotes best ucast next-hop\n\'**\' denotes best mcast next-hop\n\'[x/y]\' denotes [preference/metric]\n\'%<string>\' in via output denotes VRF <string>\n\n10.0.0.0/32, ubest/mbest: 1/0\n    *via 10.0.0.100, Po1.100, [115/6], 31w6d, isis-65535, L2\n'
'IP Route Table for VRF "test"\n\'*\' denotes best ucast next-hop\n\'**\' denotes best mcast next-hop\n\'[x/y]\' denotes [preference/metric]\n\'%<string>\' in via output denotes VRF <string>\n\n10.0.0.1/32, ubest/mbest: 1/0\n    *via 10.0.0.101, Po1.100, [115/6], 3d10h, isis-65535, L2\n'
```

cisco_nxos
```
'IP Route Table for VRF "test"\n\'*\' denotes best ucast next-hop\n\'**\' denotes best mcast next-hop\n\'\nx/y]\' denotes \npreference/metric]\n\'%<string>\' in via output denotes VRF <string>\n\n10.0.0.0/32, ubest/mbest: 1/0\n    *via 10.0.0.100, Po1.100, \n115/6], 31w6d, isis-65535, L2'
'IP Route Table for VRF "test"\n\'*\' denotes best ucast next-hop\n\'**\' denotes best mcast next-hop\n\'\nx/y]\' denotes \npreference/metric]\n\'%<string>\' in via output denotes VRF <string>\n\n10.0.0.1/32, ubest/mbest: 1/0\n    *via 10.0.0.101, Po1.100, \n115/6], 3d10h, isis-65535, L2'
```

**Comment by ktbyers** [comment]
Okay, strange...that probably is going to be hard to reproduce.

Do you have ANSI escape coloring configured on the device?


Kirk

**Comment by jesse-mcgraw** [comment]
No, nothing fancy whatsoever

Perhaps something being interpreted too broadly in strip_ansi_escape_codes?

On Tue, Feb 19, 2019 at 4:39 PM Kirk Byers <notifications@github.com> wrote:

> Okay, strange...that probably is going to be hard to reproduce.
>
> Do you have ANSI escape coloring configured on the device?
>
> Kirk
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ktbyers_netmiko_issues_1100-23issuecomment-2D465321434&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=o3FWLGO9vmXFi5VlPOamcgK_vMIETnOmBdToMYPHJrw&m=GUu94h7XbkFGFYmyGVOTpCcZ5V1vU_tjbfwpbzTWTA4&s=Wm5e4cFYKyoeXJEDGGex5_IA5m9e6JtejJKNWY844rw&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_As7p5Bd-2DyAmv61f-2DGbU5JOI6U4A9AM84ks5vPG8dgaJpZM4bDtVW&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=o3FWLGO9vmXFi5VlPOamcgK_vMIETnOmBdToMYPHJrw&m=GUu94h7XbkFGFYmyGVOTpCcZ5V1vU_tjbfwpbzTWTA4&s=uoueUeahadADE8QqH5dZYO12f7oc78ssCQAI_bmmf3k&e=>
> .
>
-- 

Jesse McGraw

Sunday - Thursday 08:00 – 04:00 ET



Enterprise Network Operations Support

KDC - 4871 Cox Road, Glen Allen, VA 23060

Phone: 800.228.5831

Email: EnterpriseNetworkOperationsSupport@capitalone.com
________________________________________________________

The information contained in this e-mail is confidential and/or proprietary to Capital One and/or its affiliates and may only be used solely in performance of work or services for Capital One. The information transmitted herewith is intended only for use by the individual or entity to which it is addressed. If the reader of this message is not the intended recipient, you are hereby notified that any review, retransmission, dissemination, distribution, copying or other use of, or taking of any action in reliance upon this information is strictly prohibited. If you have received this communication in error, please contact the sender and delete the material from your computer.

**Comment by ktbyers** [comment]
We are in process of implementing a session trace courtesy of @jobec. That will probably help with this issue.

We probably should revisit once that is in the develop branch to see if that gives us clues as to what went wrong.

**Comment by jesse-mcgraw** [comment]
Let me know if I can provide any more information

On Tue, Feb 19, 2019 at 6:25 PM Kirk Byers <notifications@github.com> wrote:

> We are in process of implementing a session trace courtesy of @jobec
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jobec&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=o3FWLGO9vmXFi5VlPOamcgK_vMIETnOmBdToMYPHJrw&m=t9o0scNy8Xo2E_4KzIBgf9Iqeb6BRVtnmEhz46yrWBE&s=z76mOEQHWiLs3scGIAjMOBoqG6QF2_KqtD2t2kw00cg&e=>.
> That will probably help with this issue.
>
> We probably should revisit once that is in the develop branch to see if
> that gives us clues as to what went wrong.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ktbyers_netmiko_issues_1100-23issuecomment-2D465353863&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=o3FWLGO9vmXFi5VlPOamcgK_vMIETnOmBdToMYPHJrw&m=t9o0scNy8Xo2E_4KzIBgf9Iqeb6BRVtnmEhz46yrWBE&s=PqpQlJAT7gz4WElnUkj1SvBn_ZK1XPxJzNZwI8nFUPU&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_As7p5LUSJI4kWJi2AGt0vTKQo0Ino6jTks5vPIfhgaJpZM4bDtVW&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=o3FWLGO9vmXFi5VlPOamcgK_vMIETnOmBdToMYPHJrw&m=t9o0scNy8Xo2E_4KzIBgf9Iqeb6BRVtnmEhz46yrWBE&s=7E8yo2c_IIxZoppZ-xhEJBcERWlHJaJLjr11sdLQeOM&e=>
> .
>
-- 

Jesse McGraw

Sunday - Thursday 08:00 – 04:00 ET



Enterprise Network Operations Support

KDC - 4871 Cox Road, Glen Allen, VA 23060

Phone: 800.228.5831

Email: EnterpriseNetworkOperationsSupport@capitalone.com
________________________________________________________

The information contained in this e-mail is confidential and/or proprietary to Capital One and/or its affiliates and may only be used solely in performance of work or services for Capital One. The information transmitted herewith is intended only for use by the individual or entity to which it is addressed. If the reader of this message is not the intended recipient, you are hereby notified that any review, retransmission, dissemination, distribution, copying or other use of, or taking of any action in reliance upon this information is strictly prohibited. If you have received this communication in error, please contact the sender and delete the material from your computer.

**Comment by carlniger** [comment]
@jesse-mcgraw can you let us know if this is still an issue on 2.3.3 or better yet the current develop branch? If it is, could you post up the session log please? [session log](https://github.com/ktbyers/netmiko/blob/develop/COMMON_ISSUES.md#enable-netmiko-logging-of-all-reads-and-writes-of-the-communications-channel)

**Comment by jesse-mcgraw** [comment]
Give me a few days and I will check

On Fri, May 24, 2019 at 5:49 PM Carl Montanari <notifications@github.com>
wrote:

> @jesse-mcgraw
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jesse-2Dmcgraw&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=olFsxTGOCEWwi99c-PCiZF2852iC4-QUy4rOWRqtPgA&m=5ZcC6O0Dspld37dejTdGC3BYQKTMkY3yeW7jNZOW7xg&s=qN1VXHfTfTCeCioM88yhE9AO7Qe-dfduO6zoiVRoc7o&e=>
> can you let us know if this is still an issue on 2.3.3 or better yet the
> current develop branch? If it is, could you post up the session log please? session
> log
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ktbyers_netmiko_blob_develop_COMMON-5FISSUES.md-23enable-2Dnetmiko-2Dlogging-2Dof-2Dall-2Dreads-2Dand-2Dwrites-2Dof-2Dthe-2Dcommunications-2Dchannel&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=olFsxTGOCEWwi99c-PCiZF2852iC4-QUy4rOWRqtPgA&m=5ZcC6O0Dspld37dejTdGC3BYQKTMkY3yeW7jNZOW7xg&s=lseWE_MNHaq58SN-Bb-6V7gxuOmSyBXzBSWlqW8p7k0&e=>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ktbyers_netmiko_issues_1100-3Femail-5Fsource-3Dnotifications-26email-5Ftoken-3DALHOTZCRFTTG2XJFDV3E77DPXBPG3A5CNFSM4GYO2VLKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWGUICI-23issuecomment-2D495797257&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=olFsxTGOCEWwi99c-PCiZF2852iC4-QUy4rOWRqtPgA&m=5ZcC6O0Dspld37dejTdGC3BYQKTMkY3yeW7jNZOW7xg&s=IreHSOrvtD2ZbgdB0RxMfGmrGRV7h-V7iIAX7TbL5Yw&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ALHOTZGXYHVOIMELQFIZTVTPXBPG3ANCNFSM4GYO2VLA&d=DwMCaQ&c=pLULRYW__RtkwsQUPxJVDGboCTdgji3AcHNJU0BpTJE&r=olFsxTGOCEWwi99c-PCiZF2852iC4-QUy4rOWRqtPgA&m=5ZcC6O0Dspld37dejTdGC3BYQKTMkY3yeW7jNZOW7xg&s=45soR4mJw5HtLmGBBesfFi5cm3zu52qCxRUj1-k0zsY&e=>
> .
>
-- 

Jesse McGraw

Sunday - Thursday 08:00 – 04:00 ET



Enterprise Network Operations Support

KDC - 4871 Cox Road, Glen Allen, VA 23060

Phone: 800.228.5831

Email: EnterpriseNetworkOperationsSupport@capitalone.com
________________________________________________________

The information contained in this e-mail is confidential and/or proprietary to Capital One and/or its affiliates and may only be used solely in performance of work or services for Capital One. The information transmitted herewith is intended only for use by the individual or entity to which it is addressed. If the reader of this message is not the intended recipient, you are hereby notified that any review, retransmission, dissemination, distribution, copying or other use of, or taking of any action in reliance upon this information is strictly prohibited. If you have received this communication in error, please contact the sender and delete the material from your computer.

**Comment by jesse-mcgraw** [comment]
This does appear to be fixed in 2.3.3
